### PR TITLE
dependency updates

### LINKED ISSUE #69: update dependencies for match newest Nov 2023
The alignment function of Biopython 1.75 and 1.81 is vastly different. The script should be updated to match the newest version.

### PR DESCRIPTION
**Pull request by DegnKristine**
fix #69 to update code to newest dependency versions. 
MAINTENANCE PR.

### CODE REVIEW

**Comment by DegnKristine** [comment]
Note, https://biopython.org/docs/latest/api/Bio.Align.Applications.html the alignment command with be removed in the future, they suggest using `import MuscleCommandline` directly.

**Comment by DegnKristine** [comment]
Tested 13 feb 2024 with biopython 1.83 - this is still viable.

File: PDBminer [hunk 1]
```diff
@@ -651,34 +653,48 @@ def alignment(uniprot_sequence, AA_pdb):
...
-        modified_string = ''
-        for i in range(len(pdb_aligned)):
-            if i < first or i > last:
-                modified_string += '-'
-            else:
-                modified_string += pdb_aligned[i]
-        
-    return uniprot_aligned, pdb_aligned, modified_string
+    if alignments.score < peptide_min_length: 
+        return [], []
```
Comment (mtiberti) [inline PDBminer:658]: maybe you could return `None` instead? It's usually used as a flag value to mean "no results"

File: PDBminer [hunk 2]
```diff
@@ -976,7 +950,8 @@ def align_uniprot_pdb(pdb_id, uniprot_sequence, uniprot_numbering, mut_pos,
         df_align, mutations_in_all, warnings = processing_normal_alignment(uniprot_sequence, 
                                                                            list(chain_df.residue_name), 
                                                                            uniprot_numbering_copy, 
-                                                                           list(chain_df.b_factor))
+                                                                           list(chain_df.b_factor),
```
Comment (mtiberti) [inline PDBminer:953]: it's usually more robust to use the `pandas` API to turn something into a list, i.e.

```
chain_df.b_factor.to_list()
```

**Review by mtiberti — CHANGES_REQUESTED** [review]

**Review by mtiberti — APPROVED** [review]

**Pull request merged**